pkg/common/dashboard.go

```diff
@@ -0,0 +1,136 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/cnrancher/autok3s/pkg/settings"
+	k3dutil "github.com/k3d-io/k3d/v5/cmd/util"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const HelmDashboardCommand = "helm-dashboard"
+
+func SwitchDashboard(ctx context.Context, enabled string) error {
+	if enabled == "true" {
+		// check if cluster list is empty
+		clusters, err := DefaultDB.ListCluster("")
+		if err != nil {
+			return err
+		}
+		if len(clusters) == 0 {
+			return errors.New("cannot enable helm-dashboard with empty cluster list")
+		}
+	}
+	if err := CheckCommandExist(HelmDashboardCommand); err != nil {
+		return err
+	}
+
+	// command execution validate
+	if err := checkDashboardCmd(); err != nil {
+		return err
+	}
+
+	isEnabled := settings.HelmDashboardEnabled.Get()
+	if !strings.EqualFold(isEnabled, enabled) {
+		if err := settings.HelmDashboardEnabled.Set(enabled); err != nil {
+			return err
+		}
+		if enabled == "true" {
+			dashboardPort := settings.HelmDashboardPort.Get()
+			if dashboardPort == "" {
+				freePort, err := k3dutil.GetFreePort()
+				if err != nil {
+					return fmt.Errorf("failed to get free port for helm-dashboard: %v", err)
+				}
+				err = settings.HelmDashboardPort.Set(strconv.Itoa(freePort))
+				if err != nil {
+					return err
+				}
+				dashboardPort = strconv.Itoa(freePort)
+			}
+			dashboardCtx, cancel := context.WithCancel(ctx)
+			DashboardCanceled = cancel
+
+			go func(ctx context.Context, p string) {
+				StartHelmDashboard(ctx, p)
+			}(dashboardCtx, dashboardPort)
+		} else {
+			if err := settings.HelmDashboardEnabled.Set("false"); err != nil {
+				return err
+			}
+			if err := settings.HelmDashboardPort.Set(""); err != nil {
+				return err
+			}
+			logrus.Info("Shutting down helm-dashboard...")
+			DashboardCanceled()
+		}
+	}
+
+	return nil
+}
+
+func InitDashboard(ctx context.Context) {
```

Comment:
Similar logic in `SwitchDashboard` function. Maybe we can reuse this function.

Reply:
fixed